CONTRIBUTING.md

```diff
@@ -1,88 +1,76 @@
-
-```sh
-make test
+```hcl
+terraform {
+    required_providers {
+        auth0 = {
+            source  = "auth0/auth0"
+            version = "0.2.0"
+        }
+    }
+}
 ```
 
-In order to run the full suite of Acceptance tests, the following environment
-variables must be set:
+## Running tests
 
-```sh
-AUTH0_DOMAIN=<your-auth0-tenant-domain>
-AUTH0_CLIENT_ID=<your-auth0-client-id>
-AUTH0_CLIENT_SECRET=<your-auth0-client-secret>
-```
+To run the tests use the `make test` command. This will make use of the pre-recorded http interactions found in the
+[recordings](./auth0/testdata/recordings) folder. To add new recordings run the tests against an Auth0 tenant
+individually using the following env var `AUTH0_HTTP_RECORDINGS=on`.
 
-Then, run `make testacc`. 
+To run the tests against an Auth0 tenant, use the `make test-e2e` command. Start by creating an
+[M2M app](https://auth0.com/docs/applications/set-up-an-application/register-machine-to-machine-applications) in the
+tenant, that has been authorized to call the Management API and has all the required permissions.
 
-**Note:** The acceptance tests make calls to a real Auth0 tenant, and create
-real resources. Certain tests also require a paid Auth0 subscription to be able to
-run successfully, e.g. `TestAccCustomDomain` and the ones starting with `TestAccLogStream*`.
+Then set the following environment variables:
 
-**Note:** At the time of writing, the following configuration steps are also
-required for the test tenant:
+* `AUTH0_DOMAIN`: The **Domain** of the M2M app
+* `AUTH0_CLIENT_ID`: The **Client ID** of the M2M app
+* `AUTH0_CLIENT_SECRET`: The **Client Secret** of the M2M app
+* `AUTH0_DEBUG`: Set to `true` to call the Management API in debug mode, which dumps the HTTP requests and responses to
+  the output
 
-* The `Username-Password-Authentication` connection must have _Requires
-  Username_ option enabled for the user tests to successfully run.
+**Note:** The e2e acceptance tests make calls to a real Auth0 tenant, and create real resources. Certain tests also
+require a paid Auth0 subscription to be able to run successfully, e.g. `TestAccCustomDomain` and the ones starting with
+`TestAccLogStream*`.
+
+**Note:** At the time of writing, the following configuration steps are also required for the test tenant:
+
+* The `Username-Password-Authentication` connection must have _Requires Username_ option enabled for the user tests to
+  successfully run.
 
 ## Documentation
 
-To make it easier to document new resources a handy script is available. The
-script can output documentation of a resource in Markdown format, using the 
-schema of the resource itself.
+To make it easier to document new resources a handy script is available. The script can output documentation of a
+resource in Markdown format, using the schema of the resource itself.
 
 ```sh
 go run scripts/gendocs.go -resource auth0_action
 ```
-
-## Releasing
```

Comment:
Removing this section as it doesn't make sense to have part of the contributing docs.